src/dialog-editor/components/modal/modalComponent.ts

```diff
@@ -0,0 +1,295 @@
+        _.extend(
+          this.DialogEditor.getDialogTabs()[
+            this.DialogEditor.activeTab],
+          { label: this.modalData.label,
+            description: this.modalData.description }
+        );
+        break;
+      case 'box':
+        _.extend(
+          this.DialogEditor.getDialogTabs()[
+            this.DialogEditor.activeTab].dialog_groups[
+              this.elementInfo.boxId],
+          { label: this.modalData.label,
+            description: this.modalData.description }
+        );
+        break;
+      case 'field':
+        this.DialogEditor.getDialogTabs()[
+          this.DialogEditor.activeTab].dialog_groups[
+            this.elementInfo.boxId].dialog_fields[
+              this.elementInfo.fieldId] = this.modalData;
+        break;
+      default:
+        break;
+    }
+  }
+
+  /**
+   * Delete dialog field selected in modal.
+   * @memberof ModalController
+   * @function deleteField
+   */
+  public deleteField() {
+    _.remove(
+      this.DialogEditor.getDialogTabs()[
+        this.DialogEditor.activeTab
+      ].dialog_groups[
+        this.elementInfo.boxId
+      ].dialog_fields,
+      (field: any) => field.position === this.elementInfo.fieldId
+    );
+  }
+
+  /**
+   * Add entry for radio button / dropdown select.
+   * @memberof ModalFieldController
+   * @function addEntry
+   */
+  public addEntry() {
+    this.modalData.values.push(['', '']);
+  }
+
+  /**
+   * Remove entry for radio button / dropdown select
+   * @memberof ModalFieldController
+   * @function removeEntry
+   * @param entry to remove from array
+   */
+  public removeEntry(entry: any) {
+    _.pull(this.modalData.values, entry);
+  }
+
+  /**
+   * Finds entries for the selected category.
+   * @memberof ModalController
+   * @function currentCategoryEntries
+   */
+  public currentCategoryEntries() {
+    if (ng.isDefined(this.categories)) {
+      return _.find(
+        this.categories.resources,
+        'id',
+        this.modalData.options.category_id
+      );
+    }
+  }
+
+  public showModal(options: any) {
+    options.controller = ['parent', function(parent) { this.parent = parent; }];
+    options.resolve = {
+      parent: () => this
+    };
+    options.controllerAs = 'modalCtrl';
+    options.template = `
+    <${options.component}
+      modal-data="modalCtrl.parent.modalData"
+      element-info="modalCtrl.parent.elementInfo"
+      categories="modalCtrl.parent.categories"
+      add-entry="modalCtrl.parent.addEntry"
+      remove-entry="modalCtrl.parent.removeEntry"
+      current-category-entries="modalCtrl.parent.currentCategoryEntries"
+      resolve-categories="modalCtrl.parent.resolveCategories"
+      modal-tab-is-set="modalCtrl.parent.modalTabIsSet"
+      modal-tab-set="modalCtrl.parent.modalTabSet"
+      modal-tab="modalCtrl.parent.modalTab"
+      save-modal="modalCtrl.parent.saveDialogFieldDetails"
+      uib-modal-instance="modalCtrl.parent.uibModalInstance"
+    ></${options.component}>
```

Comment:
Maybe move creating of template to separate static function and align the rows slightly.
When moving to specific function you should also describe the intention for such process. To have any component with (some of) these bindings as dialog template. In future new component can be created, such component will extend abstract modal class and can use any of these bindings out of the box.

```typescript
class ModalController {
  //...
  private static buildTemplate(componentName) {
    return `<${componentName} modal-data="modalCtrl.parent.modalData"
                              element-info="modalCtrl.parent.elementInfo"
                              categories="modalCtrl.parent.categories"
                              add-entry="modalCtrl.parent.addEntry"
                              remove-entry="modalCtrl.parent.removeEntry"
                              current-category-entries="modalCtrl.parent.currentCategoryEntries"
                              resolve-categories="modalCtrl.parent.resolveCategories"
                              modal-tab-is-set="modalCtrl.parent.modalTabIsSet"
                              modal-tab-set="modalCtrl.parent.modalTabSet"
                              modal-tab="modalCtrl.parent.modalTab"
                              save-modal="modalCtrl.parent.saveDialogFieldDetails"
                              uib-modal-instance="modalCtrl.parent.uibModalInstance">
           </${componentName}>`
  }

  public showModal(options: any) {
    //..
    options.template = ModalController.buildTemplate(options.component);
   //..
  }
}
```